pkg/controller/servicemesh/controlplane/reconciler.go

```diff
@@ -49,10 +52,30 @@ var orderedCharts = []string{
 	"istio/charts/kiali",
 }
 
+const (
+	smcpTemplatePath        = "/etc/istio-operator/templates/"
+	smcpDefaultTemplatePath = "/etc/istio-operator/default-templates/"
+	smcpDefaultTemplate     = "default"
+)
+
 func (r *ControlPlaneReconciler) Reconcile() (reconcile.Result, error) {
 	var err error
 	var ready bool
 
+	//render the ServiceMeshControlPlane templates
+	spec, err := r.updateSMCPWithTemplates(r.Instance.Spec)
+	if err != nil {
+		r.Log.Info(fmt.Sprintf("Warning: failed to render ServiceMeshControlPlane templates:%s", err))
+
+		//TODO: Remove these comments we've decided that templates are stable.
+		//updateReconcileStatus(&r.Instance.Status.StatusType, err)
+		//return reconcile.Result{}, err
+	} else {
+		r.Instance.Spec = spec
```

Comment:
We should not be modifying any user specified fields.  We should store the result in the status, as you have below and use that value when rendering the templates.

Reply:
Fixed